prov/efa/src/efa_fork_support.c

```diff
@@ -237,7 +248,7 @@ int efa_fork_support_enable_if_requested(struct fid_domain* domain_fid)
 	 * this variable was set to ON during provider init.  Huge pages for
 	 * bounce buffers will not be used if fork support is on.
 	 */
-	if (g_efa_fork_status == EFA_FORK_SUPPORT_OFF && efa_fork_support_is_enabled(domain_fid))
+	if (g_efa_fork_status == EFA_FORK_SUPPORT_OFF && efa_fork_support_is_enabled(domain_fid)==1)
```

Comment:
I think we should handle the case that `efa_fork_support_is_enabled()` return error by failing the function.

Reply:
Yes

Reply:
done